configuration/configuration_test.go

```diff
@@ -0,0 +1,831 @@
+package configuration_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"testing"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"strings"
+
+	"github.com/almighty/almighty-core/configuration"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/spf13/cast"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	tDefaultConfigurationFile           = "config.yaml"
+	tEnvironmentVariableNameConfigFile  = "ALMIGHTY_CONFIG_FILE_PATH"
+	tEnvironmentVariableValueConfigFile = "../config.yaml"
+	tInvalidConfigFilePath              = "../invalid_config.yaml"
+	tTestConfigFilePath                 = "../test/data/multiple_config.yaml"
+
+	// The following will be used to set env variables for testing.
+	tTestEnvString         = "ALMIGHTH_CONFIG_VALUE"
+	tTestEnvInt64    int64 = 12345
+	tTestEnvInt      int   = 12
+	tTestEnvDuration       = "1s"
+	tTestEnvBool           = false
+)
+
+func TestGetDefaultConfigurationFile(t *testing.T) {
+	t.Parallel()
+	assert.Equal(t, configuration.GetDefaultConfigurationFile(), tDefaultConfigurationFile)
+}
+
+func TestGetConfigurationDataSucess(t *testing.T) {
+	t.Parallel()
+	// this, in reality gets set in main
+	os.Setenv(tEnvironmentVariableNameConfigFile, tEnvironmentVariableValueConfigFile)
```

Comment:
I'd say this test only makes sense if you check below that the file `tEnvironmentVariableValueConfigFile` has been loaded. Maybe put some special value in the file and check for that. Currently you just check that you can get a new ConfigurationData object but nobody tells you that those aren't the default values which you probably didn't want to load.